### PR TITLE
avm2: Migrate all Events to AS, remove EventData.

### PR DESCRIPTION
**Pull request by adrian17**
This migrates all existing events to AS.

Example diff of event creation:

```rs
                        let mut io_error_evt = Avm2Event::new(
                            "ioError",
                            Avm2EventData::IOError {
                                text: "Error #2032: Stream Error".into(),
                                error_id: 2032,
                            },
                        );
                        io_error_evt.set_bubbles(false);
                        io_error_evt.set_cancelable(false);
                        let io_error_evt = EventObject::from_event(&mut activation, io_error_evt)?;
```
->
```rs
                        let io_error_evt_cls = activation.avm2().classes().ioerrorevent;
                        let io_error_evt = io_error_evt_cls.construct(
                            &mut activation,
                            &[
                                "ioError".into(),
                                false.into(),
                                false.into(),
                                "Error #2032: Stream Error".into(),
                                2032.into(),
                            ],
                        )?;
```
and for "simple events" one can skip jumping into AS constructor entirely:
```rs
        let mut removed_evt = Avm2Event::new("removed", Avm2EventData::Empty);
        removed_evt.set_bubbles(true);
        removed_evt.set_cancelable(false);
        let removed_evt = EventObject::from_event(&mut activation, removed_evt)?;
```
->
```rs
        let removed_evt = Avm2EventObject::bare_event(context, "removed", true, false);
```
Upsides:
- way less code
- way less duplication
- faster builds c:
- relatively trivial to add new events... as long as they don't happen to require native fields.

Downsides:
- had to make a custom method to create `EventObject`s without `construct`, to avoid constructing an `Activation`.
  - In general, this is slowly becoming a bit of a mess - `broadcast_event_object` creates an `Activation`, then passes its `.context` to `dispatch_event_object`, which creates _another_ `Activation`... ew. Maybe we should refactor the APIs a bit to have a nicer split between an "vm state" we can use at any time, and "AS interpreter" (Activation) that's only created if we actually execute bytecode...?
- will unavoidably be slightly slower, both construction and field accesses from Rust (will become better if we implement Rust-level precompiled known slot access)
- ugly at places; **especially** `EventObject::mouse_event` is ugly IMO. Opinions welcome :) And in general, passing 5+ positional args is less readable than creating an `EventData` with named fields.
- `ActionType::Event2` doesn't really play well with this; I had to limit its functionality to only create `Event` instances (no subclasses), which is enough for now, but might be an issue in the future.

### CODE REVIEW

**Review by Aaron1011 — APPROVED** [review]
The `Event2` change is unfortunate, but I agree that's it's better than refactoring everything to make an `Activation`/`UpdateContext` available.

File: core/src/avm2.rs [hunk 1]
```diff
@@ -154,17 +153,13 @@ impl<'gc> Avm2<'gc> {
     /// Dispatch an event on an object.
     ///
     /// The `bool` parameter reads true if the event was cancelled.
-    pub fn dispatch_event(
+    pub fn dispatch_event_object(
         context: &mut UpdateContext<'_, 'gc, '_>,
-        event: Event<'gc>,
+        event_object: Object<'gc>,
```
Comment (Herschel) [inline core/src/avm2.rs:158]: Bikeshedding: seems like renaming the fn is a little wordy to me, I prefer the shorter name, and it'd be obvious upon seeing the signature that it takes an Object.

File: core/src/avm2.rs [hunk 2]
```diff
@@ -207,12 +202,14 @@ impl<'gc> Avm2<'gc> {
     ///
     /// Attempts to broadcast a non-broadcast event will do nothing. To add a
     /// new broadcast type, you must add it to the `BROADCAST_WHITELIST` first.
-    pub fn broadcast_event(
+    pub fn broadcast_event_object(
```
Comment (Herschel) [inline core/src/avm2.rs:205]: ditto

File: core/src/avm2.rs [hunk 3]
```diff
@@ -154,17 +153,13 @@ impl<'gc> Avm2<'gc> {
     /// Dispatch an event on an object.
     ///
     /// The `bool` parameter reads true if the event was cancelled.
-    pub fn dispatch_event(
+    pub fn dispatch_event_object(
```
Comment (Herschel) [inline core/src/avm2.rs:156]: Bikeshedding: seems like renaming the fn is a little wordy to me, I prefer the shorter name, and it'd be obvious upon seeing the signature that it takes an Object.

Comment (adrian17) [reply core/src/avm2.rs:156]: TBH: I renamed it early to make "function not found" compiler errors into a simple TODO list of call sites to update :)
I can change it back now, if you think this naming is too explicit.

File: core/src/avm2.rs (hunk 1 above)
Comment (relrelb) [inline core/src/avm2.rs:158]: In the same fashion, this can be still named `event`.

**Review by Herschel — APPROVED** [review]
Thank you!

**Pull request merged**